File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int min_len = INT_MAX;
        int n = nums.size(), i = 0;
        int sum = 0;
        for(int j=0; j<n; j++) {
            sum += nums[j];
            while(sum >= target) {
                min_len = min(min_len,j-i+1);
                sum -= nums[i];
                i++;
            }
        }
        return (min_len == INT_MAX) ? 0 : min_len;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min_len = Integer.MAX_VALUE;
        int sum = 0;
        int i = 0, j = 0;
        int n = nums.length;
        while(j < n) {
            sum += nums[j];
            while(sum >= target) {
                min_len = Math.min(j-i+1,min_len);
                sum -= nums[i];
                i++;
            }
            j++;
        }
        return (min_len == Integer.MAX_VALUE) ? 0 : min_len;
    }
}